OOP_PHONE.cpp

```diff
@@ -0,0 +1,277 @@
+                    arr[j + 1] = arr[j];
+                    arr[j] = x;
+                }
+            }
+        }
+        return arr;
+    }
+
+    void sort(Person **arr, int count)
+    {
+        string x;
+        string y;
+        cout << "Please choose an ordering to sort: 'Ascending' or 'Descending'." << endl;
+        cin >> x;
+        if (x == "Ascending" || x == "Descending")
+        {
+            cout << "Please choose criteria: 'Name', 'Surname' or 'PhoneNumberCode'." << endl;
+            cin >> y;
+            cout << endl;
+            if (y == "Name" || y == "Surname")
+            {
+                sortName(arr, x, y, count);
+            }
+            else if (y == "PhoneNumberCode")
+            {
+                sortPhoneNumber(arr, x, count);
+            }
+            else
+            {
+                cout << endl
+                     << "Sorting failed" << endl;
+                cout << endl;
+            }
+        }
+        else
+        {
+            cout << endl
+                 << "Sorting failed" << endl;
+            cout << endl;
+        }
+    }
+};
+
+class File : public Helper
+{
+public:
+    int fileLine(string fileName)
+    {
+        fstream myFile;
+        int count = 0;
+        string line;
+
+        myFile.open(fileName, ios::in);
+        if (myFile.is_open())
+        {
+            while (getline(myFile, line))
+            {
+                count++;
+            }
+        }
+        myFile.close();
+        return count;
+    }
+
+    void readfile(string fileName)
+    {
+        fstream myFile;
+        cout << "File Structure:" << endl;
+        myFile.open(fileName, ios::in);
+        if (myFile.is_open())
+        {
+            string line;
+            while (getline(myFile, line))
+            {
+                cout << line << endl;
+            }
+            myFile.close();
+        }
+        cout << endl;
+    }
+
+    void addFile(Person **arr, string fileName, int count)
+    {
+        fstream myFile;
+        myFile.open(fileName, ios::app);
+        if (myFile.is_open())
+        {
+            string line;
+            for (int i = 0; i < count; i++)
+            {
+                myFile << arr[i]->name << " ";
+                myFile << arr[i]->surname << " ";
+                myFile << arr[i]->separator << " ";
+                myFile << arr[i]->phoneNumber << " " << endl;
+            }
+            myFile.close();
+        }
+    }
+};
+
+Person *person(string name, string surname, string separator, string phoneNumber)
+{
+    Person *obj = new Person();
+    obj->name = name;
+    obj->surname = surname;
+    obj->separator = separator;
+    obj->phoneNumber = phoneNumber;
+    return obj;
+}
+
+void validation(Person **arr, int count)
+{
+    cout << "Validation:" << endl;
+    for (int i = 0; i < count; i++)
+    {
+        if ((arr[i]->separator != ":" && arr[i]->separator != "-") && arr[i]->phoneNumber.length() != 9)
```

Comment:
make each validation as a separate function. Make sure you can use `n` number of separators not just 2. 

Phone number length is hardcoded with magic constant